test/EVMHost.cpp

```diff
@@ -193,8 +193,17 @@ void EVMHost::newTransactionFrame()
 		}
 	// Process selfdestruct list
 	for (auto& [address, _]: recorded_selfdestructs)
+	{
+		if (m_evmVersion >= langutil::EVMVersion::cancun() && !accounts[address].isCreated)
+		{
+			// EIP-6780: If SELFDESTRUCT is executed in a transaction different from the one
+			// in which it was created, we do NOT clear any data.
+			continue;
```

Comment:
Shouldn't you clear `recorded_selfdestructs` in this case too?

Reply:
True! I will fix ;)